.github/s2n_doxygen.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+set -e
+
+export DOXYGEN_VERSION="doxygen-1.9.3"
+curl -L "https://www.doxygen.nl/files/$DOXYGEN_VERSION.linux.bin.tar.gz" -o "$DOXYGEN_VERSION.tar.gz"
+tar -xvf "$DOXYGEN_VERSION.tar.gz"
+$DOXYGEN_VERSION/bin/doxygen docs/doxygen/Doxyfile
```

Comment:
Why not something like https://github.com/marketplace/actions/doxygen-action ?

Reply:
Initially I tried using that but I opted for a script for two reasons:
1. To control what version of doxygen we use (Though now I notice that we could also pin the GHA version to pin the Doxygen version).
2. I want to extend this script to fail if doxygen creates warnings. I did not notice that behavior for the above GHA but maybe I missed a way to add it? (I am not very familiar with GHAs).